src/history/hs.c

```diff
@@ -424,6 +427,24 @@ __hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, const
     WT_ERR(__wt_row_modify(cbt, &cursor->key, NULL, hs_upd, WT_UPDATE_INVALID, true));
 
 err:
+    /*
+     * If we inserted an update with no timestamp, we need to delete all history records for that
+     * key in front of where we just inserted. For timestamped tables that are occasionally getting
+     * a non-timestamped update, that means that all timestamped updates should get removed. In the
+     * case of non-timestamped tables, that means that all updates with higher transaction ids will
+     * get removed.
```

Comment:
I've got some questions about this - could we talk it over?
Specifically I'm not sure what you mean by "with higher transaction ids" - and I'm not seeing any code that limits when a tombstone is added based on transaction IDs. It seems to me as though this change is going to mean that we always remove history for tables that don't use timestamps. So we'd probably put some content in there the first reconciliation, then subsequently remove all the content and not add more?